projects/demo/src/modules/customization/dropdown-viewport/examples/2/portal/viewport-portal.directive.ts

```diff
@@ -0,0 +1,24 @@
+import {Directive, ElementRef, Inject} from '@angular/core';
+import {TuiDropdownPortalService} from '@taiga-ui/cdk';
+import {tuiAsViewport, TuiRectAccessor} from '@taiga-ui/core';
+
+import {PortalService} from './viewport-portal.service';
+
+@Directive({
+    selector: '[viewportPortal]',
+    providers: [
+        {provide: TuiDropdownPortalService, useExisting: PortalService},
```

Comment:
There's no need to create your own `PortalService`, I believe it would be enough to just provide a new instance of `TuiDropdownPortalService`

Reply:
fixed